### ISSUE TITLE
Fix: Failed to build uv_unwrapper

### ISSUE THREAD

**Issue by Mounir007** [Stability-AI/stable-fast-3d#45]
Under Python 3.12 i have been trying to run the project but kept receiving: 
 ```
 note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for uv_unwrapper
  Running setup.py clean for uv_unwrapper
Failed to build uv_unwrapper
ERROR: Could not build wheels for uv_unwrapper, which is required to install pyproject.toml-based proj                                         ects

```
A possible fix for this:

`cd /stable-fast-3d/uv_unwrapper/uv_unwrapper/csrc/`

In the file bvh.cpp, add #include <utility> at the top to ensure that std::exchange is recognized.
Your file "bvh.cpp" should start like this: 
```
#include "bvh.h"
#include "common.h"
#include <cstring>
#include <iostream>
#include <queue>
#include <tuple>
#include <utility>
namespace UVUnwrapper {
```

**Comment by christegho** [comment]
I am having the same issue when installing on mac. I get the following error:

```
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [13 lines of output]
      [<setuptools.extension.Extension('uv_unwrapper._C') at 0x1020b0580>]
      running bdist_wheel
      /Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/utils/cpp_extension.py:495: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
        warnings.warn(msg.format('we could not find ninja.'))
      running build
      running build_py
      copying uv_unwrapper/__init__.py -> build/lib.macosx-14.5-arm64-cpython-310/uv_unwrapper
      copying uv_unwrapper/unwrap.py -> build/lib.macosx-14.5-arm64-cpython-310/uv_unwrapper
      running build_ext
      building 'uv_unwrapper._C' extension
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Xpreprocessor -fopenmp -I/opt/homebrew/opt/libomp/include -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include/TH -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include/THC -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/include -I/Users/christophertegho/.pyenv/versions/3.10.8/include/python3.10 -c uv_unwrapper/csrc/bvh.cpp -o build/temp.macosx-14.5-arm64-cpython-310/uv_unwrapper/csrc/bvh.o -O3 -fdiagnostics-color=always -fopenmp -march=native -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_clang\" -DPYBIND11_STDLIB=\"_libcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1002\" -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++17
      clang: error: unsupported option '-fopenmp'
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
```

I have installed libomp but I still get the same error

**Comment by jammm** [comment]
> I am having the same issue when installing on mac. I get the following error:
> 
> ```
>   × python setup.py bdist_wheel did not run successfully.
>   │ exit code: 1
>   ╰─> [13 lines of output]
>       [<setuptools.extension.Extension('uv_unwrapper._C') at 0x1020b0580>]
>       running bdist_wheel
>       /Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/utils/cpp_extension.py:495: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
>         warnings.warn(msg.format('we could not find ninja.'))
>       running build
>       running build_py
>       copying uv_unwrapper/__init__.py -> build/lib.macosx-14.5-arm64-cpython-310/uv_unwrapper
>       copying uv_unwrapper/unwrap.py -> build/lib.macosx-14.5-arm64-cpython-310/uv_unwrapper
>       running build_ext
>       building 'uv_unwrapper._C' extension
>       clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Xpreprocessor -fopenmp -I/opt/homebrew/opt/libomp/include -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include/TH -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/lib/python3.10/site-packages/torch/include/THC -I/Users/christophertegho/.pyenv/versions/3.10.8/envs/s3fd/include -I/Users/christophertegho/.pyenv/versions/3.10.8/include/python3.10 -c uv_unwrapper/csrc/bvh.cpp -o build/temp.macosx-14.5-arm64-cpython-310/uv_unwrapper/csrc/bvh.o -O3 -fdiagnostics-color=always -fopenmp -march=native -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_clang\" -DPYBIND11_STDLIB=\"_libcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1002\" -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++17
>       clang: error: unsupported option '-fopenmp'
>       error: command '/usr/bin/clang' failed with exit code 1
>       [end of output]
> ```
> 
> I have installed libomp but I still get the same error

mac support is broken on the latest support atm. You can try an older commit `f8fe579df5785c2db114ff403e0e2cb506404fdf`

**Comment by vork** [comment]
Thanks to @jammm it is now fixed in: [a11ade4547fa0417754c230a7eb311e412a9f59c](https://github.com/Stability-AI/stable-fast-3d/commit/a11ade4547fa0417754c230a7eb311e412a9f59c)
